packages/loopring_v3/contracts/core/impl/libtransactions/AccountUpdateTransaction.sol

```diff
@@ -50,12 +57,17 @@ library AccountUpdateTransaction
     {
         // Read the account update
         AccountUpdate memory accountUpdate = readAccountUpdate(data, offset);
+        AccountUpdateAuxiliaryData memory auxData = abi.decode(auxiliaryData, (AccountUpdateAuxiliaryData));
```

Comment:
I feel really bad for asking the same question again: why not use the same way to decode aux data as for the transaction itself? I think we are paying the same gas for aux data as for DA.

Reply:
Would pad too many zeros when encoding the data to be used with `abi.decode` which we need to hash in the circuit which uses a lot of constraints! (as can be seen in this PR, removing ~40 bytes from the DA saves ~25,000 constraints, if they are zero of not doesn't matter for the circuit). And indeed explained here earlier: #1503 :)

Reply:
Oh, I actually meant the other way- do not use abi.decode even for aux data.

Reply:
Yeah sure that would work. But `abi.decode` is nice? :) Makes it easy to decode dynamically sized data and stuff, the only reason not to use it would be gas cost, but I don't think that's an issue here because the overhead of abi.decode is quite low, probably comparable with manually extracting the data.

Reply:
If the argument is correct, then we can say it's also ok to use abi.decode to decode the `block.data` inside `commitBlock` and the read function of all the XXXTransaction.sol.

Reply:
I see the points now.